toolkit/imageconfigs/packagelists/legacy-bootloaders.json

```diff
@@ -0,0 +1,6 @@
+{
+    "packages": [
+        "grub2"
```

Comment:
Should this be `grub2-pc` instead of `grub2`?

Reply:
Looks like the `grub2-pc` package "requires" the `grub2` package so I suppose this is mostly just a question/confirmation that the `grub2` package is enough to build the legacy grub bootloader.

Reply:
grub2-install tool is in grub2 package. We can add grub2-pc it will pull grub2 as a dependency. 
In post install then we have to remove both grub2-pc and grub2.

Reply:
I'm mostly concerned that the library files in `grub2-pc` may also be needed to create the legacy grub bootloader. Note that when we run `grub2-install` in the tools, we also pass `--target=i386-pc` which is why I suspect we might need those library files...

https://github.com/microsoft/CBL-Mariner/blob/8e3f3aef604e0c36e7b54294c8819279dfe89c88/toolkit/tools/imagegen/installutils/installutils.go#L1430

If you ran a full vhd build without `grub2-pc` here and things worked as expected, then I'm fine just keeping `grub2` as it will be less to clean up later 🙂

Reply:
Updated PR to include grub2-pc. During cleanup we can remove both grub2-pc and grub2 packages.